docs/adr/0001_cmsis_component_source_structure_recommendations.md

```diff
@@ -0,0 +1,117 @@
+# CMSIS component source structure recommendations
+Date: 2020-08-19
+
+## Status
+Proposed
+
+## Context
+
+[`ARM-software/CMSIS_5`](https://github.com/ARM-software/CMSIS_5) has been ported to Mbed OS. However, it is split across multiple directories and it is not immediately obvious that it comes from that directory. It also makes it difficult to create an independent CMake target can be built as a library that can be optionally be added by end users.
+
+The current proposal is to create a new directory `rtos/CMSIS_5/` that will mirror closely the structure used by `ARM-software/CMSIS_5/`. It will result in the removal of `rtos/source/TARGET_CORTEX/` and `cmsis/`.
+
+Additionally it will provide the following advantages:
+* Better maintenance of component
+* Easy creation of an independent CMake target
+* Easy source navigation that mirrors closely the upstream repository
+
+
+## Decision
+The following directory restructure is recommended:
+
+```
++-- rtos/CMSIS_5/
+|   +-- README.md
+|   +-- docs/
+|   +-- CMSIS/
+|       +-- include/
+|               +-- mbed_cmsis_conf.h   # Currently in cmsis/
```

Comment:
We could follow the component structure and add `include/cmsis`. Making the headers internal is probably more work than necessary for this PR.

Reply:
Sure. If you thin kit is necessary.
Done [here](https://github.com/ARMmbed/mbed-os/compare/db139ab7901ac21a56ecb4bc032c8d31ee3e0b2c..8369b887194ad8cd3e14b1c7f99f17067f56d7b6).